opensoundscape/signal_processing.py

```diff
@@ -580,29 +595,43 @@ def tdoa(
 
     See also: gcc() if you want the raw output of generalized cross correlation
     """
+    # check that the two signals are the same length
+
+    if (
+        len(reference_signal) - len(signal) > 1
+    ):  # allow 1 sample difference for rounding errors
+        raise ValueError("reference_signal and signal must be the same length.")
+
+    # check that the maximum delay is not longer than 1/2 the audio signal
+    audio_len = len(signal) / sample_rate
+    if max_delay >= audio_len / 2:
+        raise ValueError(
+            f"max_delay cannot be longer than 1/2 the signal. max_delay is {max_delay} seconds, signal is {audio_len} seconds long."
+        )
+    # trim the primary signal to just the central part
+    start = int(max_delay * sample_rate)
+    end = int(len(reference_signal) - max_delay * sample_rate)
+    signal = signal[start:end]
+
     # compute the generalized cross correlation between the signals
     cc = gcc(signal, reference_signal, cc_filter=cc_filter)
 
+    # filter to only the 'valid' part of the cross correlation, where the signals overlap fully
+    cc = cc[len(signal) - 1 : -len(signal) + 1]
+
+    # find max cc value
+    # if max is at 0 it indicates the signal arrives max_delay earlier than in the reference signal
+    lag = np.argmax(cc)
+
+    # convert lag to time delay
+    tdoa = (lag / sample_rate) - max_delay
```

Comment:
why don't we still use `correlation_lags` here? If `correlation_lags` doesn't provide the correct lags for this scenario, why do we still compute `lags` below?